### DISCUSSION TITLE
[Feature]: Add Native Fullscreen rendering to <Video /> Component

### DISCUSSION THREAD

**Discussion by gmantuanrosa** [TheWidlarzGroup/react-native-video#4989] [Ideas]
### What problem are you trying to solve?

I have a very specific scenario for my Brownfield app but encountered other developers that share the same implementation limitation -- not specific for this library, but more common to use React Native in a brownfield app.

I was developing a chat message feature to my native Android/iOS app and the decision was to implement using React Native, sharing the implementation details between different platforms.

The App is fully Native and a section of it would be to implement this feature, as we have some native bottom navigation, the best solution would be to implement them inside an Android Fragment or iOS Container. Because of that we had some limitations as to implement Navigation, Bottom Sheets and other features so we looked at react native libraries that are rendered in the Native Realm and not in the React Root View, for instance, react-native-true-sheet is an example where the bottom sheet gets rendered in the Native, bypassing the Fragment/Container giving the impression of a native component. This doesn't happen with react-native-bottom-sheet as it only gets rendered inside the fragment that is the RootView.

The same happened when we got to implement the Video Playback feature from a video message, react-native-video renders on the RootView creating a bad experience since only the Fragment would be displaying the video and not the entire phone screen.

### Proposed solution

I would like to propose something for this library -- that I don't know if is possible yet -- that is to run the `<Video />` component as full screen in the native realm.

So the Idea would be to render the Video component in the Native realm bypassing cases where React Native is running inside a Fragment/Container. This would imply the possibility to pass a header-like component to close the screen, or use the phone navigation gesture/button, but adding the possibility to customize it would be nice.

### Platforms

iOS, Android

### Additional context

Reference: `react-native-true-sheet` is an example where the bottom sheet gets rendered in the Native, bypassing the Fragment/Container giving the impression of a native component. This doesn't happen with `react-native-bottom-sheet` as it only gets rendered inside the fragment that is the RootView.

[a comment by an automated account is not shown]

**Comment by moskalakamil** [comment]
Hey @gmantuanrosa, that's right, react-native-bottom-sheet can't behave as you described because it's a JavaScript library without a native module. I just reviewed the Android implementation for react-native-video, and it seems possible to achieve this behavior. I haven't checked the iOS implementation yet. This is a rather uncommon use case, so we won't prioritize this feature at the moment. However, if you'd like to speed up this process, feel free to reach out to us for commercial support

**Comment by TWG-AI** [comment]
Thanks for the detailed writeup! Native fullscreen in a brownfield React Native setup is genuinely tricky — the current implementation assumes it can present from a standard root view controller, which breaks when RN is embedded in a more complex native hierarchy. We've noted this for the team.
